### DISCUSSION TITLE
MSBuild SDK for PascalABC.NET is now available

### DISCUSSION THREAD

**Discussion by ForNeVeR** [pascalabcnet/pascalabcnet#2709] [Show and tell]
So, I wanted to build some stuff using this wonderful compiler, but found that `dotnet build` doesn't handle PascalABC.NET projects out of the box.

I decided to publish the compiler binaries to nuget.org, and created an SDK for it (see the instructions [on the project page](https://github.com/ForNeVeR/PascalABC.NET.SDK)).

For now, it's all the very simple stuff (that just invokes the compiler), but I have plans (and experience) to introduce rudimentary NuGet support (eventually to be able to build a project [like this one](https://fornever.me/ru/posts/2017-05-15-pascalabc-net-and-avaloniaui.html)) using a proper modern SDK-styled project and `<PackageReference>`s.

My purpose is to create a future-proof ecosystem for my personal projects in PascalABC.NET, but I am working hard on the stuff to be useful for general PascalABC.NET programming using core .NET tools.

## A Showcase

Create a new file `Hello.pasproj`:
```xml
<Project Sdk="FVNever.PascalABC.NET.SDK/1.0.0">

    <PropertyGroup>
        <TargetFramework>net472</TargetFramework>
    </PropertyGroup>

    <ItemGroup>
        <MainCompile Include="Hello.pas" />
    </ItemGroup>

</Project>
```

And drop a `Hello.pas` file into the same file:
```pas
begin
    writeln('Hello, world!');
end.
```

And now you'll be able to build and run your project from the shell, no additional dependencies except the .NET SDK required:
```console
$ dotnet build
$ ./bin/Hello.exe
Hello, world!
```

**Comment by kant2002** [comment]
Here the sample repo which shows this approach https://github.com/kant2002/pascalabcsample
